### PR TITLE
Fix a possible memleak in cms_main

### PR DESCRIPTION
**Pull request by bernd-edlinger**

### CODE REVIEW

File: apps/cms.c [hunk 1]
```diff
@@ -628,7 +628,8 @@ int cms_main(int argc, char **argv)
                                  "recipient certificate file");
                 if (cert == NULL)
                     goto end;
-                sk_X509_push(encerts, cert);
+                if (!sk_X509_push(encerts, cert))
+                    goto end;
```
[the remark this answers is not in the document]
Comment (bernd-edlinger) [reply apps/cms.c:632]: No, a check for encerts != NULL is already done elsewhere,
and the vast majority of all sk_..._push checks look like this.

**Comment by bernd-edlinger** [comment]
Feel free to change *all* other sk_push call sites

**Comment by t8m** [comment]
It would be just better to fix sk_push() to never return -1.

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
> It would be just better to fix sk_push() to never return -1.

See #22930 for that change.

**Review by slontis — APPROVED** [review]

**Comment by t8m** [comment]
Merged to the master, 3.2, 3.1, and 3.0 branches. Thank you for your contribution.

**Pull request closed without merging**